### PR TITLE
0.3.0

### PR DESCRIPTION
**Pull request by ViridianMelody**
snowflake-telemetry-python 0.3.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-4247](https://anaconda.atlassian.net/browse/PKG-4247) 
- [Upstream repository](https://github.com/snowflakedb/snowflake-telemetry-python/tree/v0.3.0)
- [Upstream changelog/diff](https://github.com/snowflakedb/snowflake-telemetry-python/releases/tag/v0.3.0)

### Explanation of changes:

- Added upstream tests.

[PKG-4247]: https://anaconda.atlassian.net/browse/PKG-4247?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by skupr-anaconda — APPROVED** [review]
LGTM.
I've opened the issue about `setuptools` in tests https://github.com/snowflakedb/snowflake-telemetry-python/issues/21

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,18 +22,26 @@ requirements:
...
+    - opentelemetry-exporter-otlp ==1.12.0
 
 test:
+  source_files:
+    - tests
+    - src
   requires:
     - pip
+    # setuptools is included here to satisfy pip check.
+    - setuptools >=40,<66
```
Comment (boldorider4) [inline recipe/meta.yaml:36]: if this is truly required, it should be specified in the run section.
Weird that `pip check` raises this. I will do some inspection.

Comment (boldorider4) [reply recipe/meta.yaml:36]: ok, so it seems that it is truly required. In this case I would move it to the `run` section though, until SF fixes this.
It's a judgement call, though.

**Review by boldorider4 — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:36]: I couldn't find any usage of `setuptools` outside of the build process and test utils, and even the [upstream recipe](https://github.com/snowflakedb/snowflake-telemetry-python/blob/v0.3.0/anaconda/meta.yaml) doesn't include it in run.

**Comment by sfc-gh-tmonk** [comment]
> LGTM. I've opened the issue about `setuptools` in tests [snowflakedb/snowflake-telemetry-python#21](https://github.com/snowflakedb/snowflake-telemetry-python/issues/21)

Hi all, thanks for the heads up on this. The upper bound on setuptools was causing trouble in other ways. It is no longer required, and has been removed in this PR: https://github.com/snowflakedb/snowflake-telemetry-python/pull/22

setuptools is used to execute setup.py for builds of PyPI package artifacts.

**Pull request merged**